client/client_update.go

```diff
@@ -189,12 +190,19 @@ func makeUpdateCheckRequest(server string, current CurrentUpdate) (*http.Request
 		vals.Add("artifact_name", current.Artifact)
 	}
 
+	providesBody, err := json.Marshal(current.Provides)
+	if err != nil {
+		return nil, err
+	}
+
+	r := bytes.NewBuffer(providesBody)
+
 	ep := "/deployments/device/deployments/next"
 	if len(vals) != 0 {
 		ep = ep + "?" + vals.Encode()
 	}
 	url := buildApiURL(server, ep)
-	req, err := http.NewRequest(http.MethodGet, url, nil)
+	req, err := http.NewRequest(http.MethodGet, url, r)
```

Comment:
The new API call is a POST request. You need to try the POST first, and fall back to GET if you get a 404. The latter should not have the artifact_provides (Enterprise only).